Popup/Popup.js

```diff
@@ -569,58 +348,31 @@ class Popup extends Component {
 					ev.stopPropagation();
 					// set the pointer mode to false on keydown
 					Spotlight.setPointerMode(false);
-					forwardCustom('onClose')(null, this.props);
+					forwardCustom('onClose')(null, allComponentProps);
 				}
 			}
 		}
 	};
 
-	handleDismiss = (ev) => {
-		forwardCustom('onClose', () => ({detail: ev?.detail}))(null, this.props);
-	};
-
-	handlePopupHide = (ev) => {
-		forwardHide(ev, this.props);
-
-		this.setState({
-			floatLayerOpen: false,
-			activator: null
-		});
-
-		if (!ev.currentTarget || ev.currentTarget.getAttribute('data-spotlight-id') === this.state.containerId) {
-			this.spotActivator(this.state.activator);
-		}
-	};
-
-	handlePopupShow = (ev) => {
-		forwardShow(ev, this.props);
+	const handleKeyDownRef = useRef(handleKeyDown);
 
-		this.setState({
-			popupOpen: OpenState.OPEN
-		});
-
-		if (!ev.currentTarget || ev.currentTarget.getAttribute('data-spotlight-id') === this.state.containerId) {
-			this.spotPopupContent();
-		}
-	};
-
-	spotActivator = (activator) => {
-		this.paused.resume();
+	const spotActivator = useCallback(() => {
+		paused.current.resume();
 
 		// only spot the activator if the popup is closed
-		if (this.props.open) return;
+		if (open) return;
 
 		const current = Spotlight.getCurrent();
-		const containerNode = getContainerNode(this.state.containerId);
+		const containerNode = getContainerNode(containerId.current);
 		const lastContainerId = getLastContainer();
 
-		off('keydown', this.handleKeyDown);
+		off('keydown', handleKeyDownRef.current);
 
 		// if there is no currently-spotted control or it is wrapped by the popup's container, we
 		// know it's safe to change focus
 		if (!current || (containerNode && containerNode.contains(current))) {
 			// attempt to set focus to the activator, if available
-			if (!Spotlight.isPaused()) {
+			if (!Spotlight.isPaused() || !paused.current.isPaused()) {
```

Comment:
why was this addition condition needed?

Reply:
With this additional condition, the activator (component that opened the Popup) will be focused after the Popup is closed when it's used in other components like Input